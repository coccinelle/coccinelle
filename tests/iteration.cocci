virtual after_start

@initialize:ocaml@
@@

let tbl = Hashtbl.create 100

@r@
identifier fn;
position p;
@@

fn@p(...) { <+... return (ERR(...)); ...+> }

@script:ocaml depends on r@
fn << r.fn;
p << r.p;
@@

try Hashtbl.find tbl fn
with Not_found ->
  Hashtbl.add tbl fn ();
  let it = new iteration () in
  it#set_files [(List.hd p).file];
  it#add_virtual_rule After_start;
  it#add_virtual_identifier Err_function fn;
  it#register ()

@e depends on after_start exists@
identifier virtual.err_function;
expression x;
position p;
@@

x@p = err_function(...)

@script:python@
p << e.p;
@@

cocci.print_main("call", p)
