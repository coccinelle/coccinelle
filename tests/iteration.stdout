* TODO [[view:tests/iteration.c::face=ovl-face1::linb=6::colb=4::cole=5][call tests/iteration.c::6]]
